features/step_definitions/administration_steps.rb

```diff
@@ -0,0 +1,118 @@
+Given(/^I have an admin account$/) do
+  FactoryGirl.create(:admin_user)
+end
+
+When(/^I visit the admin panel url$/) do
+  visit new_admin_user_session_path
+end
+
+When(/^I enter my admin email$/) do
+  fill_in 'Email', with: 'admin@example.com'
+end
+
+When(/^I enter my admin password$/) do
+  fill_in 'Password', with: 'password'
+end
+
+Then(/^I see the admin panel$/) do
+  visit admin_root_path
+end
+
+Given(/^I do not have an admin account$/) do
+end
+
+Then(/^I see a flash notification that tell me that my email does not exist in the system$/) do
+  expect(page).to have_content("Invalid email or password.")
+end
+
+Given(/^I am logged into the admin panel$/) do
+  FactoryGirl.create(:admin_user)
+  visit new_admin_user_session_path
+  fill_in 'Email', with: 'admin@example.com'
+  fill_in 'Password', with: 'password'
+  click_on "Login"
+end
+
+Given(/^I am logged into the site$/) do
+  FactoryGirl.create(:user, email: "ruti@test.com")
+  ctoken = ActionMailer::Base.deliveries.last.body.match(/confirmation_token=\w*/)
+  visit "/users/confirmation?#{ctoken}"
+  visit root_path
+  fill_in "Email", with: "ruti@test.com"
+  fill_in "Password", with: "password"
+  click_on "Log in"
+end
+
+When(/^I visit the admin books url$/) do
+  visit admin_books_path
+end
+
+When(/^I enter the title "(.*?)"$/) do |title|
+  fill_in 'Title', with: title
+end
+
+When(/^I enter the price "(.*?)"$/) do |price|
+  fill_in "Price", with: price
+end
+
+When(/^I enter the published date "(.*?)"$/) do |date|
+  select "2016", from: "book_published_date_1i"
+  select "January", from: "book_published_date_2i"
+  select "26", from: "book_published_date_3i"
+end
```

Comment:
Great question! I looked into this a while back and came up with this:

````ruby

# Source: https://gist.github.com/lbspen/5674563
def select_date(date, options = {})
  date = Date.parse(date) if date.class == String
  field = options[:from]
  base_id = find(:xpath, ".//label[contains(.,'#{field}')]")[:for]

  year_format = options[:year_format] || '%Y'
  year = date.strftime(year_format)

  month_format = options[:month_format] || '%B'
  month = date.strftime(month_format)

  day_format = options[:day_format] || '%-d'
  day = date.strftime(day_format)

  select year,  from: "#{base_id}_1i"
  select month, from: "#{base_id}_2i"
  select day,   from: "#{base_id}_3i"
end

When(/^I select the date "([^"]*)" for "(.*?)"$/) do |in_date_str, field|
  select_date(in_date_str, from: field)
end
````

Feel free to use it.